juju/juju
# feat: allows to check what a machine should do in case of reboot

This PR improves the machines reboot logic by introducing the 'ShouldRebootOrShutdown'  function which take in account possible parent reboot status. 

However, parent machine are not allowed to have grandparent (it make no sense at the moment), which is enforced by the code.



## Checklist




## QA steps

run TU

```sh
TEST_PACKAGES="./domain/machine/state/... -count=1 -race -check.vv" make run-go-tests      
```


## Documentation changes


None

## Links



**Jira card:** [JUJU-6305](https://warthogs.atlassian.net/browse/JUJU-6305)



[JUJU-6305]: https://warthogs.atlassian.net/browse/JUJU-6305?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
